### PR TITLE
Fix dtype error independent normal distribution

### PR DESCRIPTION
**Pull request by elda27**
Please check my pull request about #1080 .
I solved this problem and modified a test.

I always cast Python integer to numpy.int32 because the value of numpy.int32 will upcast with computing Python integer on the Linux.

### CODE REVIEW

**Comment by elda27** [comment]
@googlebot I signed it!

File: tensorflow_probability/python/layers/distribution_layer.py [hunk 1]
```diff
@@ -896,7 +896,7 @@ def params_size(event_shape=(), name=None):
     with tf.name_scope(name or 'IndependentLogistic_params_size'):
       event_shape = tf.convert_to_tensor(
           event_shape, name='event_shape', dtype_hint=tf.int32)
-      return 2 * _event_size(
+      return np.int32(2) * _event_size(
```
Comment (ColCarroll) [inline tensorflow_probability/python/layers/distribution_layer.py:899]: Will tests pass without this cast (and the one on line 1014)?

Comment (elda27) [reply tensorflow_probability/python/layers/distribution_layer.py:899]: This line will fail in my environment. (I added the line)
https://github.com/tensorflow/probability/blob/58a5fcba16d89a9a17f1a83d97a2732808dd2733/tensorflow_probability/python/layers/distribution_layer_test.py#L829

Please see this issue.
https://github.com/numpy/numpy/issues/5297

I understood the 0-dim array or scalar of np.int32 will upcast np.int64 with computing python integer.
(Above issue is discussed np.float32 and np.float64 but the essence is the same.)
So I need the always cast the np.int32(2).

Comment (elda27) [reply tensorflow_probability/python/layers/distribution_layer.py:899]: Supplements of mine comment. There is the result dtype on my environment (on Python 3.6.5 and Ubuntu 20.04). 
Please try these lines on your environment too.

In [1]: import numpy as np
In [2]: np.__version__
Out[2]: '1.19.2'

In [3]: print((np.array([10, ], dtype=np.int32) * 4).dtype) # NumPy 1-d array
int32

In [4]: print((np.int32(10) * 4).dtype) # NumPy scalar
int64

In [5]: print((np.array(10, dtype=np.int32) * 4).dtype) # NumPy 0-d array
int64

**Review by ColCarroll — APPROVED** [review]

**Pull request merged**